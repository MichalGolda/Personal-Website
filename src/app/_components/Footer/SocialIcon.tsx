import Image from "next/image";

export type FooterIconProps = {
  src: string;
  alt: string;
  href: string;
};

const SocialIcon = ({ src, alt, href }: FooterIconProps) => {
  return (
    <a href={href}>
      <Image src={src} width={24} height={24} alt={alt} />
    </a>
  );
};

export const GithubSocialIcon = () => (
  <SocialIcon
    src="/github.webp"
    alt="github"
    href="https://github.com/michalgolda"
  />
);

export const LinkedInSocialIcon = () => (
  <SocialIcon
    src="/linkedin.webp"
    alt="linkedin"
    href="https://linkedin.com/michalgolda"
  />
);

export default SocialIcon;
