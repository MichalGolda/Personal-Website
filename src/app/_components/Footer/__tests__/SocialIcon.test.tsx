import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import SocialIcon from "../SocialIcon";

describe("SocialIcon component", () => {
  it("renders without erros", () => {
    const { container } = render(
      <SocialIcon src="/test.webp" alt="test" href="https://test.pl" />
    );

    const icon = container.firstChild;
    const image = icon?.firstChild;

    expect(icon).toHaveAttribute("href", "https://test.pl");

    expect(image).toHaveAttribute(
      "src",
      "/_next/image?url=%2Ftest.webp&w=48&q=75"
    );
    expect(image).toHaveAttribute("alt", "test");
    expect(image).toHaveAttribute("width", "24");
    expect(image).toHaveAttribute("height", "24");
  });

  it("renders unchanged", () => {
    const { container } = render(
      <SocialIcon src="/test.webp" alt="test" href="https://test.pl" />
    );

    expect(container).toMatchSnapshot();
  });
});
