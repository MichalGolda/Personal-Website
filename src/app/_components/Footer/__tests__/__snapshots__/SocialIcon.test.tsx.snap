// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialIcon component renders unchanged 1`] = `
<div>
  <a
    href="https://test.pl"
  >
    <img
      alt="test"
      data-nimg="1"
      decoding="async"
      height="24"
      loading="lazy"
      src="/_next/image?url=%2Ftest.webp&w=48&q=75"
      srcset="/_next/image?url=%2Ftest.webp&w=32&q=75 1x, /_next/image?url=%2Ftest.webp&w=48&q=75 2x"
      style="color: transparent;"
      width="24"
    />
  </a>
</div>
`;
